lightning_client.go

```diff
@@ -3042,3 +3049,36 @@ func (s *lightningClient) SubscribeInvoices(ctx context.Context,
 
 	return invoiceUpdates, streamErr, nil
 }
+
+type CheckMacPermRequest struct {
+	Macaroon string
+
+	Permissions []bakery.Op
+}
+
+func (s *lightningClient) CheckMacaroonPermissions(ctx context.Context,
+	req CheckMacPermRequest) (bool, error) {
+
+	permissions := make([]*lnrpc.MacaroonPermission, 0)
+	for _, perm := range req.Permissions {
+		permission := &lnrpc.MacaroonPermission{
+			Entity: perm.Entity,
+			Action: perm.Action,
+		}
+
+		permissions = append(permissions, permission)
+	}
+
+	rpcCtx := s.adminMac.WithMacaroonAuth(ctx)
+	res, err := s.client.CheckMacaroonPermissions(
+		rpcCtx, &lnrpc.CheckMacPermRequest{
+			Macaroon:    req.Macaroon,
```

Comment:
Over the RPC we should use `[]byte` (or `bytes` in proto) for the raw macaroon and also for the parameter of this method.

Reply:
@guggero ah ok, so regenerate the protobufs in lnd to take bytes rather than a string?